framework/src/media/media_player.c

```diff
@@ -150,14 +150,12 @@ play_result_t media_stop_play(void)
 	}
 
 	g_pc.state = PLAY_STOPPING;
-	PLAY_UNLOCK();
-
-	pthread_join(g_pc.pth, NULL);
-	
-	PLAY_LOCK();
 	g_pc.pth = -1;
+
 	PLAY_UNLOCK();
 
+	pthread_join(g_pc.pth, NULL);	
```

Comment:
Definitely wrong. You set ```g_pc.pth``` as -1 at above line. How do you use it in here?

Reply:
why do you change that?